### PR TITLE
doc: add release notes for 0.4.3

### PR DESCRIPTION
**Pull request by benesch**
@sploiselle @awang could you do your usual review of these? @benLogN you should sign off too as release manager. 

Unfortunately there were a few features that slipped through without docs, so I've added the docs here as well. Those will need additional review:

  * @mjibson can you review the `uuid` docs?
  * @ruchirK @frankmcsherry can you review the logical compaction docs?

🍻

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4232)

### CODE REVIEW

**Review by mjibson — APPROVED** [review]

File: doc/user/content/ops/deployment.md
Comment (sploiselle) [inline doc/user/content/ops/deployment.md:115]: I have a slight preference for excluding terminal markers from code samples so they're easier to copy/pate. They're relatively easy to add via CSS and make them non-selectable, so that's something I'll gladly take on if you don't mind removing these.

**Review by sploiselle — APPROVED** [review]

File: doc/user/content/ops/deployment.md
Comment (frankmcsherry) [inline doc/user/content/ops/deployment.md:45]: Maybe minor, but perhaps we should be explicit that the "window" is trailing the most recently available time. At least, I think some folks would expect window to type check as having bounds, rather than being a time. Perhaps just "The default compaction window is 60 seconds behind the current time".

File: doc/user/content/ops/deployment.md
Comment (frankmcsherry) [inline doc/user/content/ops/deployment.md:57]: Do we want to tell people about remedies here, like building a materialized view and selecting from that? I think a slightly more precise statement above replaces "especially if that query joins multiple sources" with "in the case that the query depends on multiple sources". It should not be a problem for single-source queries.

File: doc/user/content/ops/deployment.md
Comment (frankmcsherry) [inline doc/user/content/ops/deployment.md:61]: At the risk of adding too much detail, this is where `DIFFERENTIAL_EAGER_MERGE` shows up. The text is not wrong, but adding that env var greatly reduces the circumstances in which compaction doesn't occur (to those where complete compaction occurred within the logical compaction window).

Mostly trying to avoid introducing the idea that one should continually change their data or risk memory growth.

File: doc/user/content/release-notes.md
Comment (frankmcsherry) [inline doc/user/content/release-notes.md:60]: If these are the notes for 0.4.3, should the "(unreleased)" be struck?

**Review by frankmcsherry — APPROVED** [review]
I think these look fine to me. Logical compaction is a bit subtle, and I'm sure we'll have to finesse some wording at various moments as things change. This looks good to me with various take/leave nits.

File: doc/user/content/ops/deployment.md
Comment (benesch) [reply doc/user/content/ops/deployment.md:115]: Oh sure, done!

File: doc/user/content/ops/deployment.md
Comment (benesch) [reply doc/user/content/ops/deployment.md:45]: Makes total sense! Done.

File: doc/user/content/release-notes.md
Comment (benesch) [reply doc/user/content/release-notes.md:60]: Not yet! That will be @benLogN's job when he pulls the trigger on the release.

File: doc/user/content/ops/deployment.md
Comment (benesch) [reply doc/user/content/ops/deployment.md:57]: Adjusted to be more precise. I'm going to avoid adding the recommendation though because I'm not confident that I'll word it quite right, and I don't want to hold up the release notes! I would be more than happy if you wanted to run some edits on this after it lands though (or happy to lead the charge on that myself).

File: doc/user/content/ops/deployment.md
Comment (benesch) [reply doc/user/content/ops/deployment.md:61]: As above, definitely down to teach users about eager merges, but going to hold off on making any specific recommendation as to the value of `DIFFERENTIAL_EAGER_MERGE` in this PR. I'm not familiar with what a "good" value for this variable is. I also don't love publicizing a differential-branded environment variable (#1223). It's one thing for it to be a secret tuning knob that folks in the know can toggle, but a bit weird to publicly document. I'd advocate for adding a `--eager-compaction` CLI flag and `MATERIALIZE_EAGER_COMPACTION` env var that we then document.

Also, if it's great to have it turned on, should we have it turned on by default?

**Review by ruchirK — APPROVED** [review]
LGTM

File: doc/user/content/cli/_index.md
Comment (awang) [inline doc/user/content/cli/_index.md:109]: how about dropping "the size of"? Ie, changing it to:
> specifies the window behind the current time for which

is perhaps less precise, but it reads much better.  We make it pretty clear that it's a window size in the "value of the option" section below

**Review by awang — APPROVED** [review]
lgtm!  thanks so much for doing this

**Review by benLogN — APPROVED** [review]
Thanks for putting these together!

File: doc/user/content/cli/_index.md
Comment (benesch) [reply doc/user/content/cli/_index.md]: Ack, agreed it was awkwardly worded. I rewrote it, and I think it's better, but probably still not perfect. Feel free to go to town once this lands! I think @frankmcsherry might take a pass too.

**Pull request merged**
